tests/sql/operators/test_load_file.py

```diff
@@ -715,3 +717,129 @@ def test_load_file_should_fail_loudly(sample_dag, invalid_path, caplog):
         test_utils.run_dag(sample_dag)
     expected_error = f"File(s) not found for path/pattern '{invalid_path}'"
     assert expected_error in caplog.text
+
+
+def is_dict_subset(superset: dict, subset: dict) -> bool:
+    """
+    Compare superset and subset to check if the latter is a subset of former.
+    Note: dict1 <= dict2 was not working on multilevel nested dicts.
+    """
+    for key, val in subset.items():
+        print(key, val)
+        if isinstance(val, dict):
+            if key not in superset:
+                return False
+            result = is_dict_subset(superset[key], subset[key])
+            if not result:
+                return False
+        elif superset[key] != val:
+            return False
+    return True
+
+
+@pytest.mark.parametrize(
+    "remote_files_fixture",
+    [{"provider": "google"}],
+    indirect=True,
+    ids=["google_gcs"],
+)
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [{"database": Database.BIGQUERY, "table": Table(conn_id="bigquery")}],
+    indirect=True,
+    ids=["bigquery"],
+)
+def test_aql_load_file_optimized_path_method_called(
+    sample_dag, database_table_fixture, remote_files_fixture
+):
+    """
+    Verify the correct method is getting called for specific source and destination.
+    """
+    db, test_table = database_table_fixture
+    file_uri = remote_files_fixture[0]
+
+    # (source, destination) : {
+    #   method_path: where source is file source path and destination is database
+    # and method_path is the path to method
+    #   expected_kwargs: subset of all the kwargs that are passed to method mentioned in the method_path
+    #   expected_args:  List of all the args that are passed to method mentioned in the method_path
+    # }
+    optimised_path_to_method = {
+        ("gs", "bigquery",): {
+            "method_path": "airflow.providers.google.cloud.hooks.bigquery.BigQueryHook.insert_job",
+            "expected_kwargs": {
+                "configuration": {
+                    "load": {
+                        "sourceUris": [file_uri],
+                        "destinationTable": {"tableId": test_table.name},
+                    }
+                }
+            },
+            "expected_args": (),
+        }
+    }
```

Comment:
What is the benefit of having many different properties defined in a dictionary, as opposed to having separate variables containing the desired values?
```
source_file_location = "gs"
destination_database = "bigquery"
method_path = "airflow.providers.google.cloud.hooks.bigquery.BigQueryHook.insert_job"
expected_kwargs = {
                "configuration": {
                    "load": {
                        "sourceUris": [file_uri],
                        "destinationTable": {"tableId": test_table.name},
                    }
                }
expected_args = ()
```

Reply:
@tatiana I think variables are useful when we are reusing them, here this config is only consumed at one location.

Reply:
I just realised this is a generic method which will be reused across multiple optimised paths, @utkarsharma2 - I understand now :)